# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1436675872 14400
#      Sun Jul 12 00:37:52 2015 -0400
# Node ID 07dd8e602f4e72fd619bfd98e3777faa20979a86
# Parent  23172e987766113efae4a301b3e2481ac32bab64
Portably create version links in build tree

* src/module.mk (src/octave-cli-$(version)$(EXEEXT),
src/octave-gui-$(version)$(EXEEXT)): cd to target directory to create
version links.

diff --git a/src/module.mk b/src/module.mk
--- a/src/module.mk
+++ b/src/module.mk
@@ -207,16 +207,16 @@ remove-version-links:
 
 .PHONY: make-version-links remove-version-links
 
 ## We need these file names in the build tree because the wrapper
 ## program (main.cc) will try to invoke the versioned binaries.
 
 src/octave-cli-$(version)$(EXEEXT): src/octave-cli$(EXEEXT)
 	$(AM_V_GEN)rm -f $@ && \
-	$(LN_S) $(<F) $@
+	cd $(@D) && $(LN_S) $(<F) $(@F)
 
 src/octave-gui-$(version)$(EXEEXT): src/octave-gui$(EXEEXT)
 	$(AM_V_GEN)rm -f $@ && \
-	$(LN_S) $(<F) $@
+	cd $(@D) && $(LN_S) $(<F) $(@F)
 
 CLEANFILES += \
   $(OCTAVE_VERSION_LINKS)
